oci-unit-tests/standalone-checks_test.sh

```diff
@@ -93,9 +105,119 @@ test_all_supported_architectures_are_available()
+            # If the user has specified an image name to be tested, then
+            # we just test it.
+            if ! images_list=$(echo "${images_list}" | grep -Fw "${DOCKER_PACKAGE}"); then
+                continue
+            fi
+        fi
+        for image in ${images_list}; do
+            # Querying the AWS registry is easier, and we assume that the
+            # list of tags is the same for all registries.
+            for tag in $("${UTILSDIR}"/list-tags-for-image.sh \
+                                      --registry "aws" \
+                                      --namespace "${namespace}" \
+                                      --image "${image}" \
+                             | grep "${TAGS_REGEX}"); do
+                local all_digests_dir
+                local got_all_registries=1
+
+                all_digests_dir=$(mktemp -d)
+                trap 'rm -rf ${all_digests_dir}' 0 INT QUIT ABRT PIPE TERM
+
+                for registry in ${SUPPORTED_REGISTRIES}; do
+                    local manifest_list
+
+                    # The idea is to:
+                    #
+                    # - Create a file inside ${all_digests_dir} named
+                    #   "${image}-${tag}-${arch}" which will contain
+                    #   the list of all digests (from all registries)
+                    #   for the specific image/tag/arch combination.
+                    #
+                    # - Then, verify that all digests inside this
+                    #   specific file are the same.
+                    if ! manifest_list=$("${UTILSDIR}"/list-manifest-for-image-and-tag.sh \
+                                                      --registry "${registry}" \
+                                                      --namespace "${namespace}" \
+                                                      --image "${image}" \
+                                                      --tag "${tag}"); then
+                        echo "E: Could not obtain manifest list for ${registry} image ${namespace}/${image}:${tag}" > /dev/stderr
+                        ret=1
+                        got_all_registries=0
+                        break
+                    fi
+
+                    if [ -z "${manifest_list}" ]; then
+                        echo "E: Could not obtain manifest list for ${registry} image ${namespace}/${image}:${tag}" > /dev/stderr
+                        ret=1
+                        got_all_registries=0
+                        break
+                    fi
+
+                    for arch in ${SUPPORTED_ARCHITECTURES}; do
+                        if ! echo "${manifest_list}" \
+                                | jq -er ".manifests[] | select(.platform.architecture == \"${arch}\") | .digest" >> "${all_digests_dir}/${image}-${tag}-${arch}"; then
+                            echo "E: Could not obtain digest from manifest list for ${registry} image ${namespace}/${image}:${tag} (${arch})" > /dev/stderr
```

Comment:
Before this check, it would be nice if the file being handled here is a manifest list and not something else such as a json error or a image manifest for some specific arch.

Reply:
Right, but my understanding is that if we call the `jq` command on the supposed valid manifest list and it errors out, this means that we're dealing with a broken manifest list.  Or do you have something else in mind here?